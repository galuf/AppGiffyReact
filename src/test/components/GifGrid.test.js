import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Probando componente <GifGrid />", () => {
  const category = "pandas";

  test("Debe de mostrar el componente correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wraper = shallow(<GifGrid category={category} />);
    expect(wraper).toMatchSnapshot();
  });

  test("debe mostrar item cuando se carga imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://alex/alex.jpg",
        title: "alex",
      },
      {
        id: "ABCDE",
        url: "https://alex/alex.jpg",
        title: "alex",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wraper = shallow(<GifGrid category={category} />);

    expect(wraper).toMatchSnapshot();
    expect(wraper.find("p").exists()).toBe(false);
    expect(wraper.find("GridGifItem").length).toBe(gifs.length);
  });
});
