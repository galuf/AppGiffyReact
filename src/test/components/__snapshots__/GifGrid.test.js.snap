// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    pandas
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Probando componente <GifGrid /> debe mostrar item cuando se carga imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    pandas
     
  </h3>
  <div
    className="card-grid"
  >
    <GridGifItem
      id="ABC"
      key="ABC"
      title="alex"
      url="https://alex/alex.jpg"
    />
    <GridGifItem
      id="ABCDE"
      key="ABCDE"
      title="alex"
      url="https://alex/alex.jpg"
    />
  </div>
</Fragment>
`;
